src/GafferImageBindings/ImageWriterBinding.cpp

```diff
@@ -42,16 +44,48 @@
 
 #include "GafferImageBindings/ImageWriterBinding.h"
 
+using namespace std;
+using namespace boost::python;
 using namespace GafferImage;
 using namespace GafferDispatchBindings;
 
+struct DefaultColorSpaceFunctionCreator
+{
+	DefaultColorSpaceFunctionCreator( object fn )
+		:	m_fn( fn )
+	{
+	}
+
+	string operator()( string fileExtension )
+	{
+	
+		IECorePython::ScopedGILLock gilock;
+		string result = extract<string>( m_fn( fileExtension ) );
+		return result;
+
+	}
+	
+	private:
+
+		object m_fn;
+};
+
+static void registerDefaultColorSpace( object f )
```

Comment:
Once this is inside the anonymous namespace, you can remove the `static` prefix.